### ISSUE TITLE
Error while updating/loading wallet

### ISSUE THREAD

**Issue by moneymanolis** [cryptoadvance/specter-desktop#2104]
### Context
- I have two wallets with the same xpub (ghost machine based)
- I created a tx with one of them and sent it 
- Restart of Specter doesn't fix it

### Error and stack trace

```
Updating wallet Frankenstein wallet
[2023-01-31 11:23:19,276] ERROR in wallet: '378b5dda3bf3b30b98aba4ce4d7c4f2673c949953b98de0473605859dbcae272'
Traceback (most recent call last):
  File "/Users/heinzi/coding/repos/specter-desktop/src/cryptoadvance/specter/wallet.py", line 859, in check_utxo
    tx: WalletAwareTxItem = txlist_dict[utxo_txid]
KeyError: '378b5dda3bf3b30b98aba4ce4d7c4f2673c949953b98de0473605859dbcae272'
[2023-01-31 11:23:19,276] ERROR in wallet_manager: Failed to load utxos, KeyError: '378b5dda3bf3b30b98aba4ce4d7c4f2673c949953b98de0473605859dbcae272'
Traceback (most recent call last):
  File "/Users/heinzi/coding/repos/specter-desktop/src/cryptoadvance/specter/wallet.py", line 859, in check_utxo
    tx: WalletAwareTxItem = txlist_dict[utxo_txid]
KeyError: '378b5dda3bf3b30b98aba4ce4d7c4f2673c949953b98de0473605859dbcae272'

```

```

Traceback (most recent call last):
  File "/Users/heinzi/coding/repos/specter-desktop/src/cryptoadvance/specter/managers/wallet_manager.py", line 242, in _update
    loaded_wallet = self.WalletClass.from_json(
  File "/Users/heinzi/coding/repos/specter-desktop/src/cryptoadvance/specter/wallet.py", line 800, in from_json
    return cls(
  File "/Users/heinzi/coding/repos/specter-desktop/src/cryptoadvance/specter/wallet.py", line 169, in __init__
    self.update()
  File "/Users/heinzi/coding/repos/specter-desktop/src/cryptoadvance/specter/wallet.py", line 649, in update
    self.getdata()
  File "/Users/heinzi/coding/repos/specter-desktop/src/cryptoadvance/specter/wallet.py", line 943, in getdata
    self.check_utxo()
  File "/Users/heinzi/coding/repos/specter-desktop/src/cryptoadvance/specter/wallet.py", line 878, in check_utxo
    raise SpecterError(f"Failed to load utxos, {type(e).__name__}: {e}")
cryptoadvance.specter.specter_error.SpecterError: Failed to load utxos, KeyError: '378b5dda3bf3b30b98aba4ce4d7c4f2673c949953b98de0473605859dbcae272'
```

**Comment by moneymanolis** [comment]
Deleting the cache manually seemed to resolve it. I also recreated the wallet, so not sure what was the decisive factor.

**Comment by moneymanolis** [comment]
Sth. related also popped up in this Cypress test run:
https://api.cirrus-ci.com/v1/artifact/task/6213012617428992/cypress_screenshots/cypress/screenshots/spec_balances_amounts.js/Test%20the%20rendering%20of%20balances%20and%20amounts%20--%20Unconfirmed%20balance%20(failed).png

**Comment by k9ert** [comment]
Should be fixed in #2147 (uiux-revamp)  specifically 298eef213237ed5e859e90da4081895097ea2ee9. I guess the lock could be a bit narrower, though. Currently the lock is a class-level lock and so all wallets need to wait for that one lock although you could create the lock in the `__init__` and make it an object-attribute.

TBD: Port to master

**Comment by k9ert** [comment]
Port to master done as well with #2161
